### ISSUE TITLE
Selenium timeouts in Travis

### ISSUE THREAD

**Issue by jseppi** [18F/calc#883]
😢 

I got a seemingly random failure in `test_selenium.py` during testing of #879 (which made no frontend changes) in a TravisCI build.

```txt
frontend/tests/test_selenium.py:185: in wait_for
    return self.fail(err)
E   AssertionError: Timeout waiting for are_results_loaded
```

**Comment by toolness** [comment]
waegoijaweoihaweohiawg 😭 😭 😭 agwewegawge

**Comment by toolness** [comment]
Did some digging and found the [Sauce Labs job](https://saucelabs.com/beta/tests/8da9f1e7f7094335b5c551b7d39c0a8d/commands#3):

![Bad Gateway](https://cloud.githubusercontent.com/assets/124687/19187458/8e239068-8c59-11e6-8ec5-bd1d0c1cc135.png)

Looking at the source code for Django's `LiveServerTestCase`, I think there's a very small possibility for a race condition to occur between the test server being ready to accept requests and the tests actually running, but it's *super* small... I also haven't found any evidence of such a thing happening when searching Google. Blerg.

I'm hoping this is just a really rare hiccup that can be solved by restarting the travis job like you did. 😞

**Comment by jseppi** [comment]
It just failed again in #885 😭 

```txt
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
frontend/tests/test_selenium.py:185: in wait_for
    return self.fail(err)
E   AssertionError: Timeout waiting for are_results_loaded
---------------------------- Captured stdout setup -----------------------------
capabilities: {'tunnel-identifier': '2973.1', 'public': 'public', 'platform': 'ANY', 'browserName': 'internet explorer', 'version': '11.103', 'name': 'CALC', 'javascriptEnabled': True}
----------------------------- Captured stdout call -----------------------------
loading URL: http://127.0.0.1:8000/tests/
=============== 1 failed, 336 passed, 3 skipped in 60.07 seconds ===============
The command "py.test --ignore=frontend/tests/test_selenium.py --cov" exited with 1.
```

**Comment by toolness** [comment]
I despise everything.

**Comment by toolness** [comment]
So I guess one potential way to work around this might be to add some code to `setUpClass` that repeatedly makes selenium get some lightweight page, e.g. `/healthcheck`, until it actually succeeds. That would hopefully ensure that the pages the actual test(s) ping won't result in gateway errors.

**Comment by jseppi** [comment]
Related, but not sure if exactly the same:

<details>
<summary>Travis build on push to `develop` failed: https://travis-ci.org/18F/calc/builds/165874976</summary>
```txt
Starting Sauce Connect
$ travis_start_sauce_connect
Using temp dir /tmp/sc.hkuw
/tmp/sc.hkuw ~/build/18F/calc
Downloading Sauce Connect
--2016-10-07 17:07:40--  http://saucelabs.com/downloads/sc-4.4.0-linux.tar.gz
Resolving saucelabs.com (saucelabs.com)... 162.222.75.243
Connecting to saucelabs.com (saucelabs.com)|162.222.75.243|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4730565 (4.5M) [binary/octet-stream]
Saving to: `sc-4.4.0-linux.tar.gz'
100%[======================================>] 4,730,565   11.2M/s   in 0.4s    
2016-10-07 17:07:40 (11.2 MB/s) - `sc-4.4.0-linux.tar.gz' saved [4730565/4730565]
Extracting Sauce Connect
Waiting for Sauce Connect readyfile
7 Oct 17:07:41 - Sauce Connect 4.4.0, build 2863 e8cba87
7 Oct 17:07:41 - Using CA certificate bundle /etc/ssl/certs/ca-certificates.crt.
7 Oct 17:07:41 - Using CA certificate verify path /etc/ssl/certs.
7 Oct 17:07:41 - Starting up; pid 3135
7 Oct 17:07:41 - Command line arguments: sc-4.4.0-linux//bin/sc -i 2988.1 -f sauce-connect-ready-14894 -l /home/travis/sauce-connect.log 
7 Oct 17:07:41 - Log file: /home/travis/sauce-connect.log
7 Oct 17:07:41 - Pid file: /tmp/sc_client-2988.1.pid
7 Oct 17:07:41 - Timezone: UTC GMT offset: 0h
7 Oct 17:07:41 - Using no proxy for connecting to Sauce Labs REST API.
7 Oct 17:07:41 - Resolving saucelabs.com to 162.222.75.243 took 2 ms.
7 Oct 17:07:41 - Started scproxy on port 47624.
7 Oct 17:07:41 - Please wait for 'you may start your tests' to start your tests.
7 Oct 17:07:41 - Starting secure remote tunnel VM...
7 Oct 17:07:49 - Secure remote tunnel VM provisioned.
7 Oct 17:07:49 - Tunnel ID: fbabe0ffa3044d77bc61abcc484754a3
7 Oct 17:07:49 - Secure remote tunnel VM is now: booting
7 Oct 17:09:05 - Error bringing up tunnel VM.
7 Oct 17:09:05 - Cleaning up.
7 Oct 17:09:05 - Removing tunnel fbabe0ffa3044d77bc61abcc484754a3.
7 Oct 17:09:09 - All jobs using tunnel have finished.
7 Oct 17:09:09 - Sauce Connect could not establish a connection.
7 Oct 17:09:09 - Please check your firewall and proxy settings.
7 Oct 17:09:09 - You can also use sc --doctor to launch Sauce Connect in diagnostic mode.
7 Oct 17:09:09 - Goodbye.
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
The build has been terminated
```

**Comment by toolness** [comment]
Yeah that is totes out of our control. I hates it.

**Comment by jseppi** [comment]
Wait, so that failed build (and thus failed staging deploy) has nothing to do with our `test_selenium.py` stuff?

**Comment by toolness** [comment]
Oh, the https://github.com/18F/calc/issues/883#issuecomment-252313401 one you mentioned above is related to the [Sauce Connect tunnel](https://wiki.saucelabs.com/display/DOCS/Setting+Up+Sauce+Connect) not being able to initialize itself. That tunnel is required for our selenium tests to run properly on Travis+Sauce but it's completely independent of any code we've written (it's all being done by the [`sauce_connect` addon](https://docs.travis-ci.com/user/sauce-connect/)) so I'm personally putting it in the same bucket as those other bizarre Travis things that are out of our control, such as some weird ruby gem we've never heard of failing to install during Travis' bootstrapping process.

Basically the only way we can get rid of that is by removing the addon and adding our own code that sets up a sauce connect tunnel in a `travis_retry` sort of way, which would be a nightmare, or by abandoning sauce labs entirely. Which I guess is an option, it just makes me frustrated that sauce labs' own official travis add-on doesn't work consistently out of the box.

**Comment by jseppi** [comment]
Oh dear :\ What a nightmare. That means our staging deploys can just intermittently fail due to the sauce_connect addon. That seems...not good.

**Comment by toolness** [comment]
Oh blerggrerhhg.

I feel like we need an additional layer atop travis that just detects for stupid crap like this and auto-kicks travis as needed. We could call it "travis kicker" and make millions.
